swap/src/cli/list_sellers.rs

```diff
@@ -164,32 +165,32 @@ impl EventLoop {
                                 self.reachable_asb_address.insert(peer_id, address.clone());
                             }
                         }
-                        SwarmEvent::UnreachableAddr { peer_id, error, address, .. } => {
-                            if address == self.rendezvous_addr {
+                        SwarmEvent::OutgoingConnectionError { peer_id, error } => {
+                            if let Some(peer_id_rendevous) = peer_id {
```

Comment:
I think you are making some big changes to the logic here, not only refactoring. Not sure that was intended by you.

**Old:**
```rust
if address == self.rendezvous_addr {
    tracing::error!(
        "Failed to connect to rendezvous point at {}: {}",
        address,
        error
    );

    // if the rendezvous node is unreachable we just stop
    return Vec::new();
} else {
    tracing::debug!(
        "Failed to connect to peer at {}: {}",
        address,
        error
    );
    self.unreachable_asb_address.insert(peer_id, address.clone());

    match self.asb_quote_status.entry(peer_id) {
        Entry::Occupied(mut entry) => {
            entry.insert(QuoteStatus::Received(Status::Unreachable));
        },
        _ => {
            tracing::debug!(%peer_id, %error, "Connection error with unexpected peer")
        }
    }
}
```

New:
````rust
if let Some(peer_id_rendevous) = peer_id {
  tracing::error!(
      "Failed to connect to rendezvous point at {}: {}",
      self.rendezvous_addr,
      error
  );

  self.unreachable_asb_address.insert(peer_id_rendevous, self.rendezvous_addr.clone());

  match self.asb_quote_status.entry(peer_id_rendevous) { // <-- You moved this from the else block into the if block
      Entry::Occupied(mut entry) => {
          entry.insert(QuoteStatus::Received(Status::Unreachable));
      },
      _ => {
          tracing::debug!(%peer_id_rendevous, %error, "Connection error with unexpected peer")
      }
  }
} else {
  tracing::error!(
      "Failed to connect to rendezvous point at {}: {}",
      self.rendezvous_addr,
      error
  );
  // if the rendezvous node is unreachable we just stop
  return Vec::new(); // <-- You moved this from the if block to the else block
}
```

Reply:
Updated this to reflect the following cases: 
- failed to connect to the rendezvous, this is the case where we bail out
- failed to connect to a peer and have the peer_id in the event, record this peer_id in the unreachable list 
- failed to connect to a peer and peer_id is None, just log the error 

7838442947c471ce6deb91f19e989bd636f6f076

Reply:
Looks good to me. I wasn't sure if I missed something here, but you're spot on!